# CCI GUARDIANS User Stories
## Introduction

These are draft user stories for the human omics research data infrastructure under development as part of the GUARDIANS Program at Children's Cancer Institute of Australia. They reflect both work that CCI is already engaged in, and the ambitions of the project team and users for the feature set and functionality of the system to be delivered. Important caveats:

* This is absolutely not an exhaustive list, and is not in order of importance.

* There are varying levels of granularity here. Some are epic-level system stories. Some are nearly feature-level user stories. 

* This is a living document to be updated as more discovery comes to light.

* We plan to take a maximalist interpretation of the concept of ‘user’: as someone that interacts with the system in some capacity. Importantly, this includes the design, development, and operation of systems as well as ‘consumer-level’ interactions.

| US \# | Product Element | Applicable Persona “As a” | Feature/Function “I want” | Enablement Statement “So that” |
| :---- | :---- | :---- | :---- | :---- |
| US01 | Patient search | clinician | Get all withdrawn patients |  |
| US02 | Patient search | clinician, curation scientist | Get number of patients that have at least two tumours that have undergone WGS | Longitudinal studies. Tracking cancer progression to perform simpler comparisons and allowing for potential progression/evolution research |
| US03 | Patient search | clinician | Get all patients in ZERO2 that have IDH1/IDH2 mutations, as a ratio of all patients with CNS tumours |  |
| US04 | Patient search | clinician, curation scientist | Get methylation class and score for all tumour patients | This would be very useful from a research perspective being able to compare calls across different cancer types and subtypes. May help give a better understanding as to why certain samples classify one way or another. |
| US05 | Sample search | curation scientist | Get all samples with a specific variant | I can see the frequency of that specific variant within different cancer types or in general. |
| US06 | Sample search | curation scientist | Get all samples with reportable variants of a specific type (SNV, CNV, etc) | I can see the frequency of that specific variant within different cancer types or in general And for risk assessment prognostic \- whether good or bad \- targetability of specific variants. |
| US07 | Sample search | curation scientist | Search full analysis (full suite of data for a given patient) set by any id (tumour, normal, manifest, rnaseq, etc). | I can understand the biology of the cancer and can also help understand within the same subgroup of cancers whether there are trends \- eg. Ewings Sarcomas versus RMS \- and when patients come in for multiple screening (eg when cancer relapses or progresses) they get another another sample and go through process again, and we are trying to get a better understanding of evolution of the cancer. Also help understand predisposition \- underlying gene that may be influencing cancer \- may contribute to clinical monitoring guidelines. Having comprehensive dataset can help us push something to be reported because you've got more evidence to make a recommendation because you have the multi omic dataset. |
| US08 | Sample search | curation scientist | Get all samples (given a specific diagnosis eg DMG) with a specific combination of mutations | Having comprehensive dataset with multiple DMGs can help us push something to be reported because you've got more evidence to make a recommendation because you have the multi omic dataset. Also may be able to find new subcategories, can start to subcategorise cancer types. Also can help in research settings to potentially find coexisting mutations that may be related to treatment resistance rather than just looking at one mutation on its own...or as well the cancer may look under the microscope like a particular cancer type at the cellular level but might look molecularly like something different (eg a sarcoma). May help you find a new pathway...when you see several examples coming through. |
| US09 | Sample search | curation scientist | Get all samples (given a specific diagnosis eg DMG) with reportable germline mutations | An example of how this helps is: you may have a patient with an underlying germline variant but as they get older, that variant may increase the incidence and proliferation of a different type of cancer than normally would be indicated by the germline variant itself, because it influences the replication of cells in other words created mistakes in replication. Pleiotrophic \- gene can have multiple diseases associated with that gene, and so depending on the type of variant, location and type of it can determine a different disease. So it is important to look at germlines across all the samples and in a focal manner, but then also look at them as a subset and see what patterns emerge. Also may discover previously unknown dispositions, because you look at one germine variant and see one incidence, and think not known, but then you see it 5 or 6 times, then reassess its significance. Allows us to map different types of traits to the gene and the variants across the gene. |
| US10 | Sample search | curation scientist | Get the RNA Expression of a subset of genes | You can do things like targetabilities for specific expressions, the other thing is to correlate... Can do heat maps: ie. When we are looking at expressions of a group of genes we can start seeing patterns of pathways, but we can also see that pathway (pathway \= group of genes that creates an outcome) connected to another network of pathways...two different genes from different chromasomes come together and create a hybrid and create different types of expression patterns...then if you look at the cohort level, there's a lot of knowledge for discovery, adapting to treatment...You can also do subtyping of samples. Also, with radiation induced glioma, eg. a patient started with one type of cancer then radiation or treatment created another type. |
| US11 | Sample search | curation scientist | Get number of patients that have done WGS | This answer may apply to US11, US12 and US13. Comparing how the sample arrived and what was the outcomes were after being received. Could all the sequencing be performed was there extra for pre-clinical. Can compare sites and preservation methods. Can compare the sample sizes to what was able to be performed (WGS, WTS, Methylation and additional for Pre-clinical). |
| US12 | Sample search | curation scientist | Get all samples with both WGS and RNA | Comparing how the sample arrived and what the outcomes were after being received. Could all the sequencing be performed was there extra for pre-clinical. Can compare sites and preservation methods. Can compare the sample sizes to what was able to be performed (WGS, WTS, Methylation and additional for Pre-clinical). |
| US13 | Sample search | curation scientist | Get all samples with PDX or HTS data | Comparing how the sample arrived and what was the outcomes were after being received. Could all the sequencing be performed was there extra for pre-clinical. Can compare sites and preservation methods. Can compare the sample sizes to what was able to be performed (WGS, WTS, Methylation and additional for Pre-clinical). |
| US14 | Sample search | curation scientist | Get all samples with a specific tumour profile (z-score range, TMB range, specific mutsigs, etc) | Similar to US10 we can start to identify patterns on a large scale, potentially creating heat maps. To see if there are any additional factors that could influence mutational signatures or TMB. |
| US15 | Sample search | curation scientist | Get all samples with NTRK3 (with any gene), IHC positive (with that specific gene) and response rate. Using multi omic data and comparing it to stained samples on glass, following that gold standard, to determine if they were treated and if molecular profiling is superior to gold standard. | Comparing findings between IHC and what is found via molecular. Interesting cases when its larger genes and the probes used in FISH or IHC only bind to a part of the gene comparing to the WGS view of the gene (SMARCB1) |
| US16 | Sample search | curation scientist | Associate a single sample with all its ids | (NB this means the various IDs for the same patient such as PRISM id, ZERO id, but also sample, germline, tumour, clinical & more). I am able to always track what a sample is used in or for across external studies and internal studies. Also to figure out overlap between studies (one patient/variant/mutation seeming more common than it is, for example, because it appears in more than one study.) |
| US17 | Sample search | curation scientist | Get all samples with a specific class importance (VOUS3.8, etc) | This would help with reviewing variants helping us have a really accurate database. This will also help pull out interesting variants to perform functional studies. |
| US18 | Sample tracker |  | View the state of data loading for each sample (colour-coded) |  |
| US19 | Statistics Sample tracker |  | Get average time it takes to curate a case Time in pipeline (patient\_id with no samples) Time data loaded (patient has samples and purity is set) Time to curation meeting |  |
| US20 | Statistics |  | Get number of patients enrolled in each study |  |


## Glossary (this needs updating to include CCI roles and check it against GA4GH DSIP)
| Authorized | Granted conditional permission to use a resource (data or compute) |
| :---- | :---- |
| **Clinical research coordinator** | A Clinical Research Coordinator (CRC) manages and conducts the day-to-day activities of a clinical trial. |
| **Compute resource**  | For example: an analysis platform, managed workflow service, or general-purpose compute environment. These resources may be distinguished by: in-theory unbounded resource requirements of single-user jobs, and/or non-linear marginal cost to supply additional demand. For example, an online catalog service is not a compute resource. |
| **Compute resource provider** | Provider or administrator of a compute resource. Whether on-prem or in the cloud, ultimately they foot the bill — which may or may not be passed on to users. |
| **Data custodian** | A role responsible for one or more aspects of safe custody, transport and storage of data, and implementation of business rules and/or the technical environment to control access. Job titles with some data custodian responsibilities may include system administrator, database administrator, dev/ops or data engineer. |
| **Data steward** | A role responsible for ensuring data quality and fitness for purpose of the organization's data assets, and for implementation of governance. They ensure data quality, accuracy, and adherence to business rules and policies. |
| **Data owner** | The legal person, agency, or body that has authority and is accountable for their organisation’s data and may delegate responsibilities to a Data Steward. They have the authority to make decisions about data usage, access, and governance. |
| **Data user** | An individual or group that accesses and uses data for various purposes, such as research. |
| **General-purpose compute environment** | e.g. HPC, low-level cloud compute services e.g. AWS EC2, Batch |
| **Managed workflow service** | e.g. Seqera, Terra |
| **Researcher** | A (potential) consumer of the data and/or compute resources available in the ecoystem |

---


